tests/bayesian/experiment_suite.py

```diff
@@ -0,0 +1,226 @@
+                                                "learningMode": True,
+                                                "learningRate": learning_rate_l2,  # alpha
+                                                "activationThreshold": activation_threshold,  # used for cell activation & competition through distal segment activity
+                                                "forgetting": forgetting,
+                                                "resetProximalCounter": False,
+                                            }
+
+                                            L4Overrides = {
+                                                "noise": 1e-8,
+                                                "cellsPerColumn": cells_per_column,  # new: 4 # original 32
+                                                "columnCount": columns_count,  # new: 2048 # original: 2048
+                                                "minThreshold": min_threshold,
+                                                "learn": True,
+                                                "sampleSize": 40,
+                                                "useApicalTiebreak": use_apical_tiebreak,
+                                                "learningRate": learning_rate_l4,
+                                                "maxSegmentsPerCell": maxNumSegments,
+                                                "implementation": "Bayesian",
+                                                "seed": seed
+                                            }
+
+                                            title = 'Experiment with config \n\tL2Overrides=%s \n\tL4Overrides=%s \n\tImplementation=%s \n\tnumCC=%s' % (L2Overrides, L4Overrides, implementation, num_columns)
+                                            id = hash(title)
+                                            print "\n\n#################################\n", title
+                                            exp = L4L2Experiment(
+                                                title,
+                                                implementation=implementation,
+                                                L2RegionType="py.BayesianColumnPoolerRegion",
+                                                L4RegionType="py.BayesianApicalTMPairRegion",
+                                                numCorticalColumns=num_columns,
+                                                maxSegmentsPerCell=maxNumSegments,
+                                                seed=1,
+                                                L2Overrides=L2Overrides,
+                                                L4Overrides=L4Overrides,
+                                            )
+
+                                            with warnings.catch_warnings():
+                                                if (suppressWarnings):
+                                                    warnings.simplefilter("ignore")
```

Comment:
Does this catch any error thrown and continues with the next parameter setting, or does it only suppress warnings whereas errors still stop the script to execute? It is important to continue since some parameter settings just might not make any sense in their configuration and hence bring the execution to fail. If we don't catch this case we won't be able to go through all of the settings and to run all our experiments.

Reply:
Only suppresses warning. Yes for an error prone run, your setup is better or an additional catch phrase would be needed.
However, this works for simple testing.